### ISSUE TITLE
dim_of() after settimecontext() on segmented data returns strange offset

### ISSUE THREAD

**Issue by astechow** [MDSplus/mdsplus#1032]
steps to reproduce in TDI:
```
> opentree() for valid shot with segmented data
> _d = \VALID.NODE
> dim_of(_d)
Set_Range(2399997,0Q ...
> settimecontext(0QU,10000QU) my data is stored in nanoseconds
> _d = \VALID.NODE
> dim_of(_d)
[18446657673.70955D0,18446657673.70955D0, ...
```

the data seems to be returned in seconds now instead, with all the intervals still correct but this huge offset on top of everything.

**Comment by GabrieleManduchi** [comment]
I tried to reproduce the issue but everything appears to work as expected. Are you sure you are running the latest stable version? There was a similar issue about one year ago, but it had been fixed.
In case, could you provide me a pulse file reproducing the issue?

**Comment by zack-vii** [comment]
I am working on this issue part time. the issue occurs if the recorded timestamps start with negative values. i created a branch including a test the investigates this behavior but i could not fix it yet.
 zack-vii:zck_timecontext pr #1060

**Comment by zack-vii** [comment]
should be fixed in version with PR #1060

**Comment by zack-vii** [comment]
it still seems to have an offset of 86400 (i.e. second in one day) due to an incorrect setting of isFloat. This will be/is fixed though in PR #1083
